src/database/DataTypes/Item.js

```diff
@@ -284,12 +284,16 @@ export class Item extends Realm.Object {
    * uses all ItemBatches related to this item to find the values.
    * @param {Location} location
    */
-  getTemperatureExposure(location) {
-    let { batches } = this;
-    if (location) batches = this.getBatchesInLocation(location);
+  getTemperatureExposure(database, location) {
+    let sensorLogs = location ? location.getSensorLogs(database) : database.objects('SensorLog');
+    sensorLogs = sensorLogs.filtered(
+      'itemBatches.item.id = $0 && itemBatches.numberOfPacks > 0',
+      this.id
+    );
+
     return {
-      maxTemperature: batches.max('sensorLog.temperature') || -Infinity,
-      minTemperature: batches.min('sensorLog.temperature') || Infinity,
+      maxTemperature: sensorLogs.max('temperature') || -Infinity,
```

Comment:
Looks like I misspelled `sensorlog` -> `sensorLogs`. Did `batches.max('sensorLogs.temperature')` not work either? I remember testing this type of query.

Either way I think this will return the incorrect value when the `temperature` is `0`

Reply:
sure, will change

looks like max and min only works on immidiate properties.